### PR TITLE
Fixes #5320 - Timeline: timeZone and browserTimeZone are ignored when manually set

### LINKED ISSUE #5320: Timeline: timeZone and browserTimeZone are ignored when manually set
I have an issue using Timeline component in different timezones: we want to force the visualization of the events in an specific timezone, but if we force the timezone in Timeline component, is completely ignored (both _timeZone_ and _browserTimeZone_ parameters). BTW I use Primefaces 7.0.8.

The issue can be simply reproduced using any timeline demo like this: https://www.primefaces.org/showcase/ui/data/timeline/allEvents.xhtml

```
<p:timeline id="timeline" value="#{allEventsTimelineView.model}"
            editable="true" eventMargin="10" eventMarginAxis="0" 
            start="#{allEventsTimelineView.start}" 
            end="#{allEventsTimelineView.end}" 
            timeZone="Europe/Paris"
            browserTimeZone="Europe/Paris">
```

Start the application and access to the app through another device with another timezone on its computer (like American or Asian timezone). You will see that all events will be displaced in time as they are displayed with the device timezone instead of the server timezone as specified.

### PR DESCRIPTION
**Pull request by jasonex7**
Fixes #5320

### CODE REVIEW

File: src/main/java/org/primefaces/component/timeline/TimelineBase.java [hunk 1]
```diff
@@ -571,6 +572,7 @@ public void setDropScope(String dropScope) {
         getStateHelper().put(PropertyKeys.dropScope, dropScope);
     }
 
+    @Override
     public String getDir() {
         return (String) getStateHelper().eval(PropertyKeys.dir, "ltr");
     }
```
Comment (melloware) [inline src/main/java/org/primefaces/component/timeline/TimelineBase.java:587]: You can remove getDir() entirely as RTLAware interface is a default interface that does exactly this right?

**Review by melloware — CHANGES_REQUESTED** [review]

**Review by melloware — APPROVED** [review]

**Comment by christophs78** [comment]
I think we should bring back browserTimeZone (or clientTimeZone when we look at Schedule) for this. IMO the timeZone used server-side should be decoupled from the clientTimeZone. (eg we want to set server-side timeZone but do not want to set a fixed clientTimeZone)

**Comment by melloware** [comment]
we should standardize the name like clientTimeZone for both Schedule and Timeline

**Comment by jasonex7** [comment]
So `clientTimeZone` will affect vis Timeline component and `timeZone` will affect TimelineEvent start and end dates, is that correct?

**Comment by christophs78** [comment]
Thanks, looks good. 
(server-side) timezone is the "anchor". 
We send `start: new Date('2019-11-14T14:00:00+01:00'), end: new Date('2019-11-14T22:00:00+01:00')` from the server to the client.
OOTB the client (=browsers) transfers this into his (client)TimeZone. As an exmple: you want to know when the formular 1 race start. In this case you expect the browser to transfers this into your timezone. So you know when to turn on the tv.
When you attend an conference in madrid and look on the website of this converence you expect to see the starting-time in the (client)TimeZone of Madrid. For this usecase you would set clientTimeZone.

**Comment by christophs78** [comment]
@melloware : IMO this PR is ready to be merged.

**Comment by tandraschko** [comment]
all right
please also change the migration guide

**Pull request merged**